### PR TITLE
celf: init at 2013-07-25

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/logic/celf/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  name = "${pname}-${pversion}";
+
+  src = fetchFromGitHub {
+    owner  = "clf";
+    repo   = pname;
+    rev    = "d61d95900ab316468ae850fa34a2fe9488bc5b59";
+    sha256 = "0slrwcxglp0sdbp6wr65cdkl5wcap2i0fqxbwqfi1q3cpb6ph6hq";
+  };
+
+  nativeBuildInputs = [ ];
```
Comment (xeji) [inline pkgs/applications/science/logic/celf/default.nix]: This line can be omitted.

File: pkgs/applications/science/logic/celf/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    sml < main-export.sml
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp .heap* $out/bin/
+    ./.mkexec ${smlnj}/bin/sml $out/bin celf
+  '';
+
+  meta = {
```
Comment (xeji) [inline pkgs/applications/science/logic/celf/default.nix]: If you write `meta = with stdenv.lib; {` here, you don't have to repeat `stdenv.lib.` 3 times below.

**Comment by bcdarwin** [comment]
@xeji -- both fixed.

**Comment by xeji** [comment]
@GrahamcOfBorg build celf

**Comment by xeji** [comment]
Thank you!

**Pull request merged**